src/Serialization/Fields.jl

```diff
@@ -48,3 +48,102 @@ function load_internal(s::DeserializerState, ::Type{gfp_fmpz_elem}, dict::Dict)
     F = load_type_dispatch(s, Nemo.GaloisFmpzField, dict[:parent])
     return F(load_type_dispatch(s, fmpz, dict[:data]))
 end
+
+################################################################################
+# SimpleNumField
+function save_internal(s::SerializerState,
+                       K::SimpleNumField{T}) where T
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K)),
+        :var => save_type_dispatch(s, String(var(K)))
+    )
+end
+ 
+function load_internal(s::DeserializerState,
+                       ::Type{<: SimpleNumField{T}},
+                       dict::Dict) where T
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    var = load_type_dispatch(s, dict[:var], check_namespace=false)
+    K, _ = NumberField(def_pol, var, cached=false)
+    
+    return K
+end
+
+################################################################################
+# Fqnmodfinitefield
+function save_internal(s::SerializerState,
+                       K::FqNmodFiniteField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{FqNmodFiniteField},
+                       dict::Dict) where T
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    K, _ = FiniteField(def_pol, cached=false)
+    
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState,
+                       k::Union{nf_elem, fq_nmod, Hecke.NfRelElem{T}}) where T
+    K = parent(k)
+    polynomial = parent(defining_polynomial(K))(k)
+    K_dict = save_type_dispatch(s, K)
+    
+    return Dict(
+        :parent => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{nf_elem, fq_nmod, Hecke.NfRelElem{T}}},
+                       dict::Dict) where T
+    K = load_type_dispatch(s, dict[:parent], check_namespace=false)
+    polynomial = load_type_dispatch(s, dict[:polynomial], check_namespace=false)
+    return K(polynomial)
+end
+
+################################################################################
+# NfAbsNS 
+function save_internal(s::SerializerState, K::Union{NfAbsNS})
+    def_pols = defining_polynomials(K)
+    pols_parent = parent(def_pols[1])
+    return Dict(
+        :pols_parent => save_type_dispatch(s, pols_parent),
+        :def_pols => save_type_dispatch(s, def_pols),
+        :vars => save_type_dispatch(s, [String(a) for a in vars(K)])
```

Comment:
Just wondering: why not simply
```suggestion
        :vars => save_type_dispatch(s, vars(K))
```

Reply:
because they're symbols, and on load they need to be strings, I end up removing the possibility of storing the strings used in non simple extensions for now since the constructor only a accepts a string and not a vector of strings. I could try to pick out the string from the first variable, but i feel that may sometimes lead to errors if depending on the type of string used, and maybe it's best to allow the constructor to allow a vector of strings? I am not sure here

Reply:
I think it's a missing feature that those constructors don't accept symbols: because they internally store symbols (so why force me to create strings which then are converted to symbols anyway?), and because in similar constructors we allow both strings and symbols. It would be great if you could file bug reports for these cases, ideally as explicit as possible: state which type is missing the constructor and give a minimal example how to reproduce this. (Of course even better would be to make PRs to add those missing constructors ;-) ).

Reply:
I just created this https://github.com/thofma/Hecke.jl/pull/709